config_src/infra/FMS1/MOM_io_infra.F90

```diff
@@ -750,44 +750,55 @@ subroutine read_field_4d(filename, fieldname, data, MOM_Domain, &
   ! needed.
   ! call read_data(filename, fieldname, data, MOM_Domain%mpp_domain, &
   !                timelevel=timelevel, position=position)
 
+  use_fms_read_data = .true. ; if (present(file_may_be_4d)) use_fms_read_data = .not.file_may_be_4d
   file_is_global = .true. ; if (present(global_file)) file_is_global = global_file
 
-  if (file_is_global) then
-    call mpp_open(unit, trim(filename), form=NETCDF_FILE, action=READONLY_FILE, &
-                  threading=MULTIPLE, fileset=SINGLE_FILE) !, domain=MOM_Domain%mpp_domain )
+  use_fms_read_data = .false. ! 4d arrays not working with FMS1
+
+  if (use_fms_read_data) then
+    call read_data(filename, fieldname, data, MOM_Domain%mpp_domain, &
+                   timelevel=timelevel, position=position)
   else
-    call mpp_open(unit, trim(filename), form=NETCDF_FILE, action=READONLY_FILE, &
-                  threading=MULTIPLE, fileset=MULTIPLE, domain=MOM_Domain%mpp_domain )
-  endif
-  call mpp_get_info(unit, ndim, nvar, natt, ntime)
-  allocate(fields(nvar))
-  call mpp_get_fields(unit, fields(1:nvar))
-  do n=1, nvar
-    call mpp_get_atts(fields(n), name=varname)
-    if (lowercase(trim(varname)) == lowercase(trim(fieldname))) then
-        call MOM_error(NOTE, "Reading 4-d variable "//trim(fieldname)//" from file "//trim(filename))
-      ! Maybe something should be done depending on the value of ntime.
-      call mpp_read(unit, fields(n), MOM_Domain%mpp_domain, data, timelevel)
-      exit
+    if (file_is_global) then
+      call mpp_open(unit, trim(filename), form=NETCDF_FILE, action=READONLY_FILE, &
+                    threading=MULTIPLE, fileset=SINGLE_FILE) !, domain=MOM_Domain%mpp_domain )
+    else
+      call mpp_open(unit, trim(filename), form=NETCDF_FILE, action=READONLY_FILE, &
+                    threading=MULTIPLE, fileset=MULTIPLE, domain=MOM_Domain%mpp_domain )
     endif
-  enddo
-  if ((n == nvar+1) .or. (nvar < 1)) call MOM_error(WARNING, &
-    "read_field apparently did not find 4-d variable "//trim(fieldname)//" in file "//trim(filename))
+    call mpp_get_info(unit, ndim, nvar, natt, ntime)
+    allocate(fields(nvar))
+    call mpp_get_fields(unit, fields(1:nvar))
+    do n=1, nvar
+      call mpp_get_atts(fields(n), name=varname)
+      if (lowercase(trim(varname)) == lowercase(trim(fieldname))) then
+          call MOM_error(NOTE, "Reading 4-d variable "//trim(fieldname)//" from file "//trim(filename))
+        ! Maybe something should be done depending on the value of ntime.
+        call mpp_read(unit, fields(n), MOM_Domain%mpp_domain, data, timelevel)
+        exit
+      endif
+    enddo
+    if ((n == nvar+1) .or. (nvar < 1)) call MOM_error(WARNING, &
+      "read_field apparently did not find 4-d variable "//trim(fieldname)//" in file "//trim(filename))
 
-  deallocate(fields)
-  call mpp_close(unit)
+    deallocate(fields)
+    call mpp_close(unit)
+  endif
```

Comment:
As we discussed, this change to read_field_4d probably not required, since we are not using `fms_field_may_be_4d`, `use_fms_read_data` is hard-coded to false, and we are moving most of our testing to FMS2.

Since the proposed solution relies on the `global_file` flag, we should be good to revert the change to this file.  (In fact, I almost wonder if all of the `read_field_*` functions should use `global_file`, but that is a conversation for another day.)